datascience/tables.py

```diff
@@ -753,7 +746,7 @@ class instance. The formatter is run each time ``__repr__`` is called. So, while
         """
         if inspect.isclass(formatter):
             formatter = formatter()
-        if callable(formatter) and not hasattr(formatter, 'format_column'):
+        if not hasattr(formatter, 'format_column') and callable(formatter):
```

Comment:
Can we revert this properly to ensure that the order of the checks is the same as before? This could cause issues that would've been averted due to Python short-circuiting.

Reply:
Sorry for the delay, yes, I have reverted this change.